src/utils/flow.js

```diff
@@ -36,14 +36,61 @@ export function immutableUpdate<T: Object>(object: T, ...rest: Object[]): T {
  * This function takes a string and returns either a valid TabSlug or null, this doesn't
  * throw an error so that any arbitrary string can be converted, e.g. from a URL.
  */
-export function toValidTabSlug(tabSlug: string): TabSlug | null {
-  switch (tabSlug) {
+export function toValidTabSlug(tabSlug: any): TabSlug | null {
```

Comment:
With `any` Flow won't tell you if you try to pass, I don't know, an object, or a number, which could indicate a bug.
Currently it's string | void (not null).
I won't fight for this but I still think this would be better.